pkgs/by-name/im/immich/package.nix

```diff
@@ -0,0 +1,316 @@
+{
+  lib,
+  pkgs,
+  buildNpmPackage,
+  fetchFromGitHub,
+  python3,
+  nodejs,
+  runCommand,
+  nixosTests,
+  # build-time deps
+  pkg-config,
+  makeWrapper,
+  cmake,
+  curl,
+  cacert,
+  unzip,
+  # runtime deps
+  ffmpeg-full, # immich requires at least webp encoder from ffmpeg-full
+  imagemagick,
+  libraw,
+  libheif,
+  vips,
+  perl,
+}:
+let
+  buildNpmPackage' = buildNpmPackage.override { inherit nodejs; };
+  sources = lib.importJSON ./sources.json;
+  inherit (sources) version;
+
+  meta = with lib; {
+    description = "Self-hosted photo and video backup solution";
+    homepage = "https://immich.app/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ jvanbruegge ];
+    inherit (nodejs.meta) platforms;
+  };
+
+  buildLock = {
+    sources =
+      builtins.map
+        (p: {
+          name = p.pname;
+          inherit (p) version;
+          inherit (p.src) rev;
+        })
+        [
+          imagemagick
+          libheif
+          libraw
+        ];
+
+    packages = [ ];
+  };
+
+  # The geodata website is not versioned, so we use the internet archive
+  geodata =
+    runCommand "immich-geodata"
+      {
+        outputHash = "sha256-imqSfzXaEMNo9T9tZr80sr/89n19kiFc8qwidFzRUaY=";
+        outputHashMode = "recursive";
+        nativeBuildInputs = [ cacert ];
+
+        meta.license = lib.licenses.cc-by-40;
+      }
```

Comment:
You mean split this into 5 different derivations just to combine them again at the end? I don't really see any benefit. Just requires juggling 5 different sha hashes instead of one